sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/implementation/AzureNettyHttpClientContext.java

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.netty.implementation;
+
+import com.azure.core.util.ProgressReporter;
+
+/**
+ * A holder for all context that may be passed through the Netty HttpClient.
+ */
+public final class AzureNettyHttpClientContext {
```

Comment:
`responseTimeoutOverride` and `progressReporter` are already in the context. Do we need a new type to hold these two values?

Reply:
Reduces the number of lookups we need to do in `Context`, so if we add more configurations in the future they all can be stored in one location.

Reply:
The number of lookups should be the same if we do the lookup in the ctor of AzureSdkHandler (and the context passed in as ctor arg). Will not require us to create a new instance of this type each time in a hot path.